ssas/connection.go

```diff
@@ -1,35 +1,53 @@
 package ssas
 
 import (
+	"database/sql"
 	"os"
-	"runtime"
+	"time"
 
 	_ "github.com/lib/pq"
-	log "github.com/sirupsen/logrus"
 	"gorm.io/driver/postgres"
 	"gorm.io/gorm"
 )
 
-// Variable substitution to support testing.
-var LogFatal = log.Fatal
+var Connection *gorm.DB
+
+func init() {
+	var err error
+	Connection, err = createDB()
 
-func GetGORMDbConnection() *gorm.DB {
-	databaseURL := os.Getenv("DATABASE_URL")
-	db, err := gorm.Open(postgres.Open(databaseURL), &gorm.Config{})
 	if err != nil {
-		LogFatal(err)
+		Logger.Fatalf("Failed to create db %s", err.Error())
 	}
-	return db
 }
 
-func Close(db *gorm.DB) {
-	d, err := db.DB()
+func createDB() (*gorm.DB, error) {
+	databaseURL := os.Getenv("DATABASE_URL")
+
+	sqlDB, err := sql.Open("pgx", databaseURL)
+	if err != nil {
+		return nil, err
+	}
+
+	// db.SetMaxOpenConns(cfg.MaxOpenConns)
+	// db.SetMaxIdleConns(cfg.MaxIdleConns)
+	// db.SetConnMaxLifetime(time.Duration(cfg.ConnMaxLifetimeMin) * time.Minute)
+
+	sqlDB.SetMaxOpenConns(40)
+	sqlDB.SetMaxIdleConns(40)
+	sqlDB.SetConnMaxLifetime(time.Duration(5) * time.Minute)
+
+	db, err := gorm.Open(postgres.New(postgres.Config{
```

Comment:
`postgres.New` and `postgres.Config` are only supported in newer versions of the gorm postgres driver, so I had to upgrade. This also seems to force an upgrade of gorm.